pxtblocks/blocklycompiler.ts

```diff
@@ -1454,8 +1463,9 @@ namespace pxt.blocks {
         const variableIsScoped = (b: Blockly.Block, name: string): boolean => {
             if (!b)
                 return false;
-            else if ((b.type == "controls_for" || b.type == "controls_simple_for" || b.type == "controls_for_of")
-                && escapeVarName(b.getField("VAR").getText(), e) == name)
+            else if ((b.type == "controls_for" || b.type == "controls_simple_for" || b.type == "controls_for_of"
```

Comment:
We should move all these types into a constant array or something

Reply:
Sure